src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/DecryptorPalWindows.Decrypt.cs

```diff
@@ -0,0 +1,223 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Win32.SafeHandles;
+
+using static Interop.Crypt32;
+
+namespace Internal.Cryptography.Pal.Windows
+{
+    internal sealed partial class DecryptorPalWindows : DecryptorPal
+    {
+        public sealed override ContentInfo TryDecrypt(RecipientInfo recipientInfo, X509Certificate2 cert, X509Certificate2Collection originatorCerts, X509Certificate2Collection extraStore, out Exception exception)
+        {
+            Debug.Assert(recipientInfo != null);
+            Debug.Assert(cert != null);
+            Debug.Assert(originatorCerts != null);
+            Debug.Assert(extraStore != null);
+
+            CryptKeySpec keySpec;
+            exception = TryGetKeySpecForCertificate(cert, out keySpec);
```

Comment:
It feels really weird that you have a Try method (return data, or return null-and-out-Exception) which calls a Try method that returns an exception or outs data.

A) Why aren't they just throwing?
B) Why the paradigm flip?